src/main/java/org/opensearch/knn/plugin/transport/GetModelTransportAction.java

```diff
@@ -27,17 +29,25 @@ public class GetModelTransportAction extends HandledTransportAction<GetModelRequ
     private static final Logger LOG = LogManager.getLogger(GetModelTransportAction.class);
     private ModelDao modelDao;
 
+    private final Client client;
+
     @Inject
-    public GetModelTransportAction(TransportService transportService, ActionFilters actionFilters) {
+    public GetModelTransportAction(TransportService transportService, ActionFilters actionFilters, Client client) {
         super(GetModelAction.NAME, transportService, actionFilters, GetModelRequest::new);
         this.modelDao = ModelDao.OpenSearchKNNModelDao.getInstance();
+        this.client = client;
     }
 
     @Override
     protected void doExecute(Task task, GetModelRequest request, ActionListener<GetModelResponse> actionListener) {
-        String modelID = request.getModelID();
-
-        modelDao.get(modelID, actionListener);
+        // temporary setting thread context to default, this is needed to allow actions on model system index when security plugin is
+        // enabled
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
```

Comment:
Same as above. Isnt this redundant also, given threadcontext is already stashed in modelDao?